src/modules/EventModule.cpp

```diff
@@ -0,0 +1,463 @@
+#include "core/Permissions.h"
+#include "core/Utility.h"
+#include "CurrencyModule.h"
+
+#include <QtCore/QFile>
+#include <QtCore/QJsonDocument>
+#include <QtCore/qregexp.h>
+#include <QtCore/QtMath>
+
+#include <random>
+
+#define eventConfigLoc QString("eventConfig")
+
+EventModule::EventModule(UmikoBot* client) : Module("event", true), m_client(client)
+{
+	RegisterCommand(Commands::EVENT, "event", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+		auto& config = getEventServerData(channel.guildId());
+		CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+		auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+
+		if (args.size() != 1)
+		{
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		if (config.isEventRunning == false)
+		{
+			client.createMessage(message.channelId(), "**No event is active at this moment.**");
+			return;
+		}
+		if (config.eventHighRiskHighRewardRunning)
+		{
+			int num = config.eventTimer->remainingTime();
+			QString time = utility::StringifyMilliseconds(num);
+
+			Discord::Embed embed;
+			embed.setColor(15844367);
+			embed.setTitle("**HighRiskHighReward** event");
+			embed.setDescription("The steal chance is **decreased** but if you succeed you will get **bonus " + currencyConfig.currencySymbol + "**!\n"
+				"Event **expires** in `" + time + "`");
+
+			client.createMessage(message.channelId(), embed);
+			return;
+
+		}
+		if (config.eventLowRiskLowRewardRunning)
+		{
+			int num = config.eventTimer->remainingTime();
+			QString time = utility::StringifyMilliseconds(num);
+
+			Discord::Embed embed;
+			embed.setColor(15844367);
+			embed.setTitle("**LowRiskLowReward** event");
+			embed.setDescription("The steal chance is **increased** but you have to give a **penalty** if you succeed!\n"
+				"Event **expires** in `" + time + "`");
+
+			client.createMessage(message.channelId(), embed);
+			return;
+		}
+
+	});
+	RegisterCommand(Commands::EVENT_GIVE_NEW_ACCESS, "give-new-event-access", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+		{
+			QStringList args = message.content().split(' ');
+
+			Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN, [this, args, message, channel](bool result)
+				{
+					GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+					if (!result)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**You don't have permissions to use this command.**");
+						return;
+					}
+					if (args.size() < 2)
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**Wrong Usage of Command!**");
+						return;
+					}
+					snowflake_t guild = channel.guildId();
+					snowflake_t chan = channel.id();
```

Comment:
This is never used.

Reply:
The `guild` is used, but the `chan` is not, I am removing the `chan` 🙂